contracts/misc/KyberInternalGovernance.sol

```diff
@@ -0,0 +1,137 @@
+import {IKyberGovernance} from '../interfaces/governance/IKyberGovernance.sol';
+import {IRewardsDistributor} from '../interfaces/rewardDistribution/IRewardsDistributor.sol';
+
+
+/**
+    Internal contracts to participate in KyberDAO and claim rewards for Kyber
+*/
+contract KyberInternalGovernance is PermissionAdmin, PermissionOperators, Utils {
+    using SafeERC20 for IERC20Ext;
+
+    address payable public rewardRecipient;
+    IKyberGovernance public governance;
+    IRewardsDistributor public rewardDistributor;
+
+    event ClaimedRewards(uint256 cycle, uint256 index, IERC20Ext[] tokens, uint256[] claimAmounts);
+
+    constructor(
+        address _admin,
+        address payable _rewardRecipient,
+        IKyberGovernance _governance,
+        IRewardsDistributor _rewardDistributor,
+        address _operator
+    ) PermissionAdmin(_admin) {
+        require(_rewardRecipient != address(0), "invalid reward recipient");
+        require(_governance != IKyberGovernance(0), "invalid kyber governance");
+        require(_rewardDistributor != IRewardsDistributor(0), "invalid reward distributor");
+
+        rewardRecipient = _rewardRecipient;
+        governance = _governance;
+        rewardDistributor = _rewardDistributor;
+
+        if (_operator != address(0)) {
+            // consistent with current design
+            operators[_operator] = true;
+            operatorsGroup.push(_operator);
+            emit OperatorAdded(_operator, true);
+        }
+    }
+
+    receive() external payable {}
+
+    /**
+    * @dev only operator can vote, given list of proposal ids
+    *   and an option for each proposal
+    */
+    function vote(
+        uint256[] calldata proposalIds,
+        uint256[] calldata optionBitMasks
+    )
+        external onlyOperator
+    {
+        require(proposalIds.length == optionBitMasks.length, "invalid length");
+        for(uint256 i = 0; i < proposalIds.length; i++) {
+            governance.submitVote(proposalIds[i], optionBitMasks[i]);
+        }
+    }
+
+    /**
+    * @dev anyone can call to claim rewards for multiple epochs
+    * @dev all eth will be sent back to rewardRecipient
+    */
+    function claimRewards(
+        uint256 cycle,
+        uint256 index,
+        IERC20Ext[] calldata tokens,
+        uint256[] calldata cumulativeAmounts,
+        bytes32[] calldata merkleProof
+    )
+        external
+        returns (uint256[] memory claimAmounts)
+    {
+        claimAmounts = rewardDistributor.claim(
+            cycle, index, address(this), tokens, cumulativeAmounts, merkleProof
+        );
+
+        for(uint256 i = 0; i < tokens.length; i++) {
+            uint256 bal = tokens[i] == ETH_TOKEN_ADDRESS ?
+                address(this).balance : tokens[i].balanceOf(address(this));
+            if (bal > 0) _transferToken(tokens[i], bal);
+        }
+        
+        emit ClaimedRewards(cycle, index, tokens, claimAmounts);
```

Comment:
this event is duplicated in RewardDistributor
and there is the case when another people call claim reward for this contract